src/classify/kernel/radial_basis.cpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file radial_basis.cpp
+ * @author Chase Geigle
+ */
+
+#include "classify/kernel/radial_basis.h"
+#include "io/packed.h"
+
+namespace meta
+{
+namespace classify
+{
+namespace kernel
+{
+
+const util::string_view radial_basis::id = "rbf";
+
+radial_basis::radial_basis(double gamma) : gamma_{gamma}
+{
+    // nothing
+}
+
+radial_basis::radial_basis(std::istream& in)
+{
+    io::packed::read(in, gamma_);
+}
+
+void radial_basis::save(std::ostream& out) const
+{
+    io::packed::write(out, id);
+    io::packed::write(out, gamma_);
+}
+
+double radial_basis::operator()(const feature_vector& first,
+                                const feature_vector& second) const
+{
+    auto first_it = std::begin(first);
+    auto first_end = std::end(first);
+
+    auto second_it = std::begin(second);
+    auto second_end = std::end(second);
+
+    auto dist = 0.0;
+    while (first_it != first_end && second_it != second_end)
+    {
+        if (first_it->first == second_it->first)
+        {
+            auto delta = first_it->second - second_it->second;
+            dist += delta * delta;
+            ++first_it;
+            ++second_it;
+        }
+        else if (first_it->first < second_it->first)
+        {
+            dist += first_it->second * first_it->second;
+            ++first_it;
+        }
+        else
+        {
+            dist += second_it->second * second_it->second;
+            ++second_it;
+        }
+    }
+
+    for (; first_it != first_end; ++first_it)
```

Comment:
I don't think that will make it any more clear, since you'd need to tack on additional checks to the branches of the if/else chain to ensure that e.g. `first_it` or `second_it` is even valid to look at.